src/scripts/object.search.js

```diff
@@ -96,32 +81,20 @@ module.exports = (robot, res) => {
 		processObjectSearch(robot, res, res.match[1]);
 	});
 
-	function enableClassifierCleanupInterval() {
-		if (nlcHelper.initializedSuccessfully()) {
-			context.robot.logger.debug('NLC enabled so classifier will take place');
-			setInterval(() => {
-				nlcHelper.deleteOldClassifiers()
-					.then((result) => {
-						robot.logger.debug('Successfully deleted old classifiers.  ' + JSON.stringify(result, null, 2));
-					})
-					.catch((err) => {
-						robot.logger.error('Problem deleting old classifiers.', err);
-					});
-			}, env.nlc_classifier_cleanup_interval);
-		}
-		else {
-			context.robot.logger.debug('NLC not enabled so classifier cannot take place');
-		}
-	}
-
-	function searchForObject(context, nlcHelper, searchPhrase) {
-		return nlcHelper.classify(searchPhrase)
+	function searchForObject(context, searchPhrase) {
+		return env.searchEngine.classify(searchPhrase)
 			.then((classifierResult) => {
 				let matches = [];
-				if (classifierResult.classes) {
+				if (!classifierResult.search_successful || !classifierResult.classify_result) {
+					// no result returned
+					let error = classifierResult.description ? new Error(classifierResult.description) : new Error(classifierResult);
+					context.robot.logger.error(`${TAG}: No classifiers are available at this time.`, error);
```

Comment:
I was thinking that here is a good spot to emit a message to the robot response.  Essentially if search_successful is false, description is guaranteed to have a translated message that is suitable to show to the end user (within the bot client).  Without this it will be hard for the user to know why the search didn't work.